drivers/SmartThings/zigbee-switch/src/robb/init.lua

```diff
@@ -35,36 +29,15 @@ local function is_robb_dimmer(opts, driver, device)
   return false
 end
 
-local function energy_meter_handler(driver, device, value, zb_rx)
-  local raw_value = value.value
-  local multiplier = device:get_field(constants.SIMPLE_METERING_MULTIPLIER_KEY) or 1
-  local divisor = device:get_field(constants.SIMPLE_METERING_DIVISOR_KEY) or 1000000
-
-  raw_value = raw_value * multiplier / divisor
-  device:emit_event(capabilities.energyMeter.energy({ value = raw_value, unit = "kWh" }))
-end
-
-local function power_meter_handler(driver, device, value, zb_rx)
-  local raw_value = value.value
-  local multiplier = device:get_field(constants.ELECTRICAL_MEASUREMENT_MULTIPLIER_KEY) or 1
-  local divisor = device:get_field(constants.ELECTRICAL_MEASUREMENT_DIVISOR_KEY) or 10
-
-  raw_value = raw_value * multiplier / divisor
-  device:emit_event(capabilities.powerMeter.power({ value = raw_value, unit = "W" }))
+local do_init = function(driver, device)
+  device:set_field(constants.SIMPLE_METERING_DIVISOR_KEY, 1000000, {persist = true})
+  device:set_field(constants.ELECTRICAL_MEASUREMENT_DIVISOR_KEY, 10, {persist = true})
 end
 
 local robb_dimmer_handler = {
-  NAME = "ROBB smarrt dimmer",
-  zigbee_handlers = {
-    attr = {
-      [SimpleMetering.ID] = {
-        [SimpleMetering.attributes.InstantaneousDemand.ID] = power_meter_handler,
```

Comment:
The default logic for 'ActivePower' is fine but 
in case of 'InstantaneousDemand', the value from the default logic would be '(raw_value / 1000)' but this handler is expecting '(raw_value / 10)'

Reply:
good catch